### ISSUE TITLE
datawarp I/O being filtered out due to its location in /var [REPLACEMENT ISSUE]

### ISSUE THREAD

**Issue by shanedsnyder** [darshan-hpc/darshan#216]
The original issue

	Id: 216
	Title: datawarp I/O being filtered out due to its location in /var

could not be created.
This is a dummy issue, replacing the original one. It contains everything but the original issue description. In case the gitlab repository is still existing, visit the following link to show the original issue:

TODO

**Comment by shanedsnyder** [comment]
In GitLab by @carns on Jan 16, 2017, 15:56

I think we should just add a whitelist.  If a match is found (before return(1)) then we consult the whitelist and give it a chance to reverse the decision.

**Comment by shanedsnyder** [comment]
In GitLab by @glennklockwood on Jan 17, 2017, 14:41

Agreed, this sounds good.   I can take a stab at it next week.

**Comment by shanedsnyder** [comment]
In GitLab by @glennklockwood on Jan 24, 2017, 17:07

This whitelist feature now allows us to selectively watch directories of interest which were previous forbidden such as `/var/tmp` and `/dev/shm`.  Should I add these to the whitelist along with the DataWarp mount, or just leave it as narrow as possible?

I mostly ask because some apps (and middleware) use these as alternatives to `/tmp`, which is not blacklisted.

**Comment by shanedsnyder** [comment]
In GitLab by @carns on Jan 24, 2017, 17:16

Sure, that makes sense to me.

**Comment by shanedsnyder** [comment]
In GitLab by @glennklockwood on Jan 24, 2017, 17:28

After some basic testing, the only downside is that it appears that many MPI implementations use /dev/shm for shared memory sorts of stuff.  For example, Cray MPI generates the following:

```
POSIX	-1	3939887926310529934	POSIX_BYTES_WRITTEN	0	/dev/shm/cray-shared-mem-coll-kvs_90003589850-2-3-69615.tmp	/dev	devtmpfs
POSIX	-1	9754444816381394869	POSIX_BYTES_WRITTEN	0	/dev/shm/cray-shared-mem-coll-kvs_90003589850-1-1-69615.tmp	/dev	devtmpfs
POSIX	-1	14847915557612568047	POSIX_BYTES_WRITTEN	0	/dev/shm/cray-shared-mem-coll-kvs_90003589850-2-2-69615.tmp	/dev	devtmpfs
```

and I know MVAPICH does something similar.  This might make things confusing for users who don't purposely use /dev/shm, which is why (I presume) the blacklist exists in the first place.  After seeing a darshan log filled with these mystery middleware files, I am having doubts about whitelisting /dev/shm and/or /var/tmp.

A future feature may be to allow the whitelist to be populated from a runtime environment variable, but that's out of scope for fixing this issue.  Strong opinions one way or another?

**Comment by shanedsnyder** [comment]
In GitLab by @carns on Jan 24, 2017, 19:49

Oh, I didn't think about that.  In that case I would suggest just leaving the datawarp path in the whitelist and not worrying about the other paths for now.  It's ok if we revisit the environment variable idea later unless you already have it coded up.

**Comment by shanedsnyder** [comment]
In GitLab by @glennklockwood on Jan 24, 2017, 20:14

mentioned in merge request !7

**Comment by shanedsnyder** [comment]
In GitLab by @shanedsnyder on Jan 25, 2017, 10:34

closed via merge request !7

**Comment by shanedsnyder** [comment]
In GitLab by @shanedsnyder on Jan 25, 2017, 10:34

closed via commit c49c9a1e8a8eb4d3dd108855bf012806acef1e40

**Comment by shanedsnyder** [comment]
In GitLab by @shanedsnyder on Feb 13, 2017, 14:56

mentioned in commit intel-hpdd/darshan-dxlt@c49c9a1e8a8eb4d3dd108855bf012806acef1e40
